### PR TITLE
feat: new CLI parameter to execute a command and quit (without CLI interaction)

### PR DESCRIPTION
**Pull request by spena**
### Description 
Fixes https://github.com/confluentinc/ksql/issues/4981

Adds a new paramter to the CLI to execute a command and quit. This without having to interact with the CLI console.
```
sergio@pop-os:~/Development/ksql$ time ./bin/ksql -e "show streams;"

 Stream Name         | Kafka Topic                 | Format 
------------------------------------------------------------
 KSQL_PROCESSING_LOG | default_ksql_processing_log | JSON   
 TOPIC1              | topic1                      | JSON   
------------------------------------------------------------

real	0m2.250s
user	0m4.604s
sys	0m0.185s
```

```
sergio@pop-os:~/Development/ksql$ ./bin/ksql -e "set 'auto.offset.reset'='earliest'; select * from topic1 emit changes;"
Successfully changed local property 'auto.offset.reset' to 'earliest'. Use the UNSET command to revert your change.
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|ID                                                                                                                                                                                                                               |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|1                                                                                                                                                                                                                                |
^CQuery terminated
```

You can use a different output format to easily parse the command result. For instance, JSON:
```
$ ./bin/ksql -e "show streams" --output JSON | jq '.[].streams' | jq '.[].name'
"TOPIC1"
"KSQL_PROCESSING_LOG"
"S1"
```

### Testing done 
Added unit tests
Verified manually

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by colinhicks** [comment]
> Adds a new paramter to the CLI to execute a command and quit. This without having to interact with the CLI console.

Just to clarify: For streaming results like push queries without a limit clause, you need to also issue ctrl+c to interrupt the console, correct?

**Comment by colinhicks** [comment]
> * Add new parameter to remove headers and columns from the command output
> * Add new parameter to change command output format (i.e. comma-separated and tab-separated)

It might be worth considering a flag to send the raw JSON to stdout. This way it could be piped to other scripts like `jq` for such formatting.

**Comment by spena** [comment]
@colinhicks Ctrl+C is part of the stream. The example of the Description shows the `^Query terminated` line that was caused by the Ctrl+C.

File: ksqldb-cli/src/main/java/io/confluent/ksql/cli/Options.java [hunk 1]
```diff
@@ -111,6 +113,12 @@
           + "defaults to TABULAR)")
   private String outputFormat = OutputFormat.TABULAR.name();
 
+  @SuppressWarnings("unused") // Accessed via reflection
+  @Option(
+      name = {EXECUTE_OPTION, EXECUTE_SHORT_OPTION},
+      description = "Execute a command and quit.")
```
Comment (rodesai) [inline ksqldb-cli/src/main/java/io/confluent/ksql/cli/Options.java:119]: nit: the user may not understand "command" - maybe "run one or more ksql statements and quit."?

File: ksqldb-cli/src/main/java/io/confluent/ksql/cli/Cli.java [hunk 2]
```diff
@@ -179,13 +182,31 @@ public void makeKsqlRequest(final String statements) {
...
+    RemoteServerSpecificCommand.validateClient(terminal.writer(), restClient);
+    try {
+      // Handles the RUN SCRIPT command if found
+      if (!terminal.maybeHandleCliSpecificCommands(command)) {
+        handleLine(command);
+      }
+    } catch (final EndOfFileException exception) {
+      // Ignore - only used by runInteractively() to exit the CLI
+    } catch (final Exception exception) {
+      LOGGER.error("", exception);
```
Comment (rodesai) [inline ksqldb-cli/src/main/java/io/confluent/ksql/cli/Cli.java:195]: why the empty log message?

**Review by rodesai — COMMENTED** [review]
LGTM other than a couple minor comments inline. Also, is there a place in the docs we can add this?

File: ksqldb-cli/src/main/java/io/confluent/ksql/cli/Options.java (hunk 1 above)
Comment (spena) [reply ksqldb-cli/src/main/java/io/confluent/ksql/cli/Options.java]: Done.

File: ksqldb-cli/src/main/java/io/confluent/ksql/cli/Cli.java (hunk 2 above)
Comment (spena) [reply ksqldb-cli/src/main/java/io/confluent/ksql/cli/Cli.java]: Done.

**Review by agavra — APPROVED** [review]
LGTM, I think @rodesai meant to check the green tick

**Pull request merged**